### ISSUE TITLE
New error seen in Homebridge console

### ISSUE THREAD

**Issue by stevehoek** [dgreif/ring#164]
I'm now seeing this in my Homebridge console:

```
System error
(node:651) UnhandledPromiseRejectionWarning: Error: This callback function has already been called by someone else; it can only be called one time.
    at /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/once.js:12:13
    at Camera.<anonymous> (/homebridge/node_modules/homebridge-ring/lib/homebridge/base-accessory.js:100:17)
    at Generator.next (<anonymous>)
    at fulfilled (/homebridge/node_modules/homebridge-ring/lib/homebridge/base-accessory.js:5:58)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

```

Any idea what is going on?
I have about 6 cameras and the views are working.  I don't get the alerts from HomeKit, but do from the Ring app.

**Comment by dgreif** [comment]
That error happens when we call a homebridge callback twice.  I just noticed I'm seeing this in my logs as well.  Will investigate...

**Comment by smockle** [comment]
Taking a look at these lines: https://github.com/dgreif/ring/blob/2e6ce703006fcd66a8964055f968c0774cda181a/homebridge/base-accessory.ts#L59-L68

Is `callback` invoked twice if `requestUpdate` throws? (First, with a value, on line 61. Second, with an error, on line 67.)

**Comment by dgreif** [comment]
@smockle that was my first guess as well.  I did some debugging yesterday and it looks like it's actually an issue with the in-home doorbell switch clashing with the siren switch.  Since they are both a `Switch` service, we need to use a `subType` to differentiate them.  This is a side effect of another subtle bug with the in-home doorbell switch thinking that non-doorbells have an in-home doorbell.  I thought everything was perfect when we merged that in, but it looks like my non-doorbell cameras have `chime_settings` with `type: 0`, which causes `hasInHomeDoorbell` to return `true`.  It seemed like it was properly filtering them out when run with homebridge because the siren switch got created before the in-home doorbell switch, and the in-home doorbell switch latched on to the existing switch service.  Hopefully that all makes sense 😅.  I should have a fix out in the next day or two
